pkg/controller/volume/persistentvolume/scheduler_binder_test.go

```diff
@@ -1380,22 +1409,53 @@ func TestBindPodVolumes(t *testing.T) {
 			},
 			shouldFail: true,
 		},
+		"node-affinity-check-fails": {
+			binding:    binding1aBoundSelectedNode,
+			cachedPVs:  []*v1.PersistentVolume{pvNode1a, pvNode2},
+			cachedPVCs: []*v1.PersistentVolumeClaim{selectedNodePVC},
+			nodes:      []*v1.Node{node1, node2},
+			delayFunc: func(t *testing.T, testEnv *testEnv, pod *v1.Pod, pv *v1.PersistentVolume, pvc *v1.PersistentVolumeClaim) {
+				// Update PVC to be fully bound to a PV with a different node
+				newPVC := pvc.DeepCopy()
+				newPVC.ResourceVersion = "100"
+				newPVC.Spec.VolumeName = pvNode2.Name
+				metav1.SetMetaDataAnnotation(&newPVC.ObjectMeta, annBindCompleted, "yes")
+
+				// Update PVC cache, fake client doesn't invoke informers
+				internalBinder, ok := testEnv.binder.(*volumeBinder)
+				if !ok {
+					t.Fatalf("Failed to convert to internal binder")
+				}
+
+				pvcCache := internalBinder.pvcCache
+				internalPVCCache, ok := pvcCache.(*pvcAssumeCache)
+				if !ok {
+					t.Fatalf("Failed to convert to internal PVC cache")
+				}
+				internalPVCCache.add(newPVC)
+			},
+			shouldFail: true,
+		},
 	}
 
 	for name, scenario := range scenarios {
 		klog.V(4).Infof("Running test case %q", name)
 
 		// Setup
 		pod := makePod(nil)
-		if scenario.apiPV == nil {
-			scenario.apiPV = scenario.cachedPV
+		if scenario.apiPVs == nil {
+			scenario.apiPVs = scenario.cachedPVs
 		}
 		testEnv := newTestBinder(t)
 		if !scenario.bindingsNil {
+			if scenario.nodes == nil {
+				scenario.nodes = []*v1.Node{node1}
+			}
+			testEnv.initNodes(scenario.nodes)
 			if scenario.binding != nil {
```

Comment:
I would add a new field to the test case for provisioned pvcs.  Then for the call to `assumeVolumes`, the bindings argument is determined by `scenario.binding`, and the provisioned pvcs argument is determined by `scenario.provisionedPVCs`.  That way, the arguments can be independent.

There can even be two test cases to test wrong binding.  One for static binding, and one for dynamic provisioning.

Reply:
Done.

> There can even be two test cases to test wrong binding. One for static binding, and one for dynamic provisioning.

Do you mean something like this?